### PR TITLE
json server refactor to dynamic js files

### LINKED ISSUE #109: refactor mock-api to allow for separate/dynamic db files
partition db.json into separate files according to API endpoints { trade, exploration, outfitting, etc }
allow for `yarn dev-api` to auto-update when db source files are changed

### PR DESCRIPTION
**Pull request by aslink87**
covers #109 
separates db.json file into module files (`trade` and `exploration` for now, `outfitting` later)
each file can include multiple endpoints
adds nodemon and glob packages to allow for realtime updates to database source files
simplifies `json-server` configs

example: developer wants to mock endpoint `/api/v1/exploration/system/by-name-containing`
1. within `mock` directory, update exploration.js file by adding a new object:
`{ 'exploration-system-by-name-containing.js': [...] }`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "dev": "next dev",
     "dev-api": "concurrently \"next dev\" \"yarn api\"",
-    "api": "node mock-api/server.js",
+    "mock": "json-server --watch ./mock-api/_db.js --routes ./mock-api/routes.json --middlewares ./mock-api/middlewares/_postAsGet.js --port 3001",
```
Comment (nitishdhar) [inline package.json]: Is this around to just run the mock API independently? Can you update the README if this changes anything?

**Comment by nitishdhar** [comment]
> example: developer wants to mock endpoint `/api/v1/exploration/system/by-name-containing`
> 
> 1. within `mock` directory, create new directories `exploration/system`
> 2. within `mock/exploration/system` directory, create new `by-name-containing.js` file to export mock data

Doesn't this add more overhead on how nested paths can be implement with just writing routes? e.g., `/api/v1/exploration/system/by-name-containing` can be just a route config pointing to `by-name-containing` object. 

Another concern I have with this approach is re-using responses on different paths, routing config can just point to the same response path.

One value I see here is how we can manage the data in splits rather than one huge file. I don't mind implementing that while retaining the routing approach rather than the folder approach. i.e., have some data folder hold different json files as needed to split the data by resource types but then have them represent one db.json file using server.js and use routing to hit the path needed. Let me know what you think of that.

Absolutely love the real-time update part, much needed.

TL;DR; - Inclined to introduce a resource specific splitting of data files to improve management, add the watch setup for db changes but retain the routing approach to pointing to different response objects over having to create new folders for each new path.

File: package.json (hunk 1 above)
Comment (nitishdhar) [reply package.json]: Nvm, I see it now

**Review by nitishdhar — APPROVED** [review]
Thanks for this improvement 👍
